### PR TITLE
Allow control over TLS version and ciphers for docker-registry

### PR DESCRIPTION
**Pull request by liggitt**
Allows envvar-based control over tls version and ciphers for docker registry via envvars

```
REGISTRY_HTTP_TLS_MINVERSION=VersionTLS10
REGISTRY_HTTP_TLS_CIPHERSUITES=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,...
```

Valid tls versions are "VersionTLS10", "VersionTLS11", "VersionTLS12"

Valid cipher suites are https://golang.org/pkg/crypto/tls/#pkg-constants for the golang version used to build origin, taken from descriptions of http://www.iana.org/assignments/tls-parameters/tls-parameters.xml#tls-parameters-4

I'm not sure whether we want to try to add config file options for these... the stanza for TLS options is part of docker standard config

follow up to https://github.com/openshift/origin/pull/13167 for the docker registry

### CODE REVIEW

**Comment by mfojtik** [comment]
@miminar @legionus FYI

**Comment by liggitt** [comment]
[test]

File: pkg/cmd/dockerregistry/dockerregistry.go [hunk 1]
```diff
@@ -36,6 +36,8 @@ import (
 	_ "github.com/docker/distribution/registry/storage/driver/s3-aws"
 	_ "github.com/docker/distribution/registry/storage/driver/swift"
 
+	"strings"
```
Comment (miminar) [inline pkg/cmd/dockerregistry/dockerregistry.go]: This should go way up.

**Comment by mfojtik** [comment]
[merge]

@miminar agreed the strings import is ok to fix later.

**Pull request merged**